packages/protocol/test-sol/devchain/e2e/common/EpochManager.t.sol

```diff
@@ -677,6 +679,205 @@ contract E2E_EpochManager_FinishNextEpochProcess is E2E_EpochManager {
+        break;
+      }
+    }
+  }
+
+ /**
+  * @notice Calculates the lesser, greater neighbors, and original index for a group after a potential vote revocation.
+  * @param targetGroup The group whose votes are being revoked.
+  * @param revokeAmount The amount of votes being revoked.
+  * @return lesser The address of the group with the next lower vote count after revocation.
+  * @return greater The address of the group with the next higher vote count after revocation.
+  * @return index The original index of the targetGroup in the eligible list before simulation.
+  */
+  function _calculateRevokeNeighbors(
+    address targetGroup,
+    uint256 revokeAmount
+  ) internal view returns (address lesser, address greater, uint index) {
+    (address[] memory groupsInOrder, GroupWithVotes[] memory groupWithVotesSimulated) = getGroupsWithVotes();
+
+    uint targetGroupIndexSimulated = groupWithVotesSimulated.length;
+    for (uint i = 0; i < groupWithVotesSimulated.length; i++) {
+      if (groupWithVotesSimulated[i].group == targetGroup) {
+        if (groupWithVotesSimulated[i].votes >= revokeAmount) {
+          groupWithVotesSimulated[i].votes -= revokeAmount;
+        } else {
+          groupWithVotesSimulated[i].votes = 0;
+        }
+        targetGroupIndexSimulated = i;
+        break;
+      }
+    }
+    require(targetGroupIndexSimulated < groupWithVotesSimulated.length, "Target group not found for revoke simulation");
+
+    sort(groupWithVotesSimulated);
+
+    lesser = address(0);
+    greater = address(0);
+    for (uint i = 0; i < groupWithVotesSimulated.length; i++) {
+      if (groupWithVotesSimulated[i].group == targetGroup) {
+        lesser = (i == groupWithVotesSimulated.length - 1) ? address(0) : groupWithVotesSimulated[i + 1].group;
+        greater = (i == 0) ? address(0) : groupWithVotesSimulated[i - 1].group;
+        break;
+      }
+    }
+
+    index = groupsInOrder.length;
+    for (uint i = 0; i < groupsInOrder.length; i++) {
+      if (groupsInOrder[i] == targetGroup) {
+        index = i;
+        break; 
+      }
+    }
+    require(index < groupsInOrder.length, "Target group not found for revoke index");
+  }
+
+  struct AliceContext {
+    address alice;
+    uint256 lockedAmount;
+    address targetGroup;
+  }
+
+  function _setupAlice() internal returns (AliceContext memory ctx) {
+    ctx.alice = vm.addr(uint256(keccak256(abi.encodePacked("alice"))));
+    vm.deal(ctx.alice, 100_000 ether);
+    ctx.lockedAmount = 1 ether;
+
+    vm.startPrank(ctx.alice);
+    accounts.createAccount();
+    lockedCelo.lock{ value: ctx.lockedAmount }();
+    vm.stopPrank();
+
+    uint256 actualLocked = lockedCelo.getAccountTotalLockedGold(ctx.alice);
+    require(actualLocked == ctx.lockedAmount, "Alice lock failed");
+
+    (, GroupWithVotes[] memory initialGroups) = getGroupsWithVotes();
+    require(initialGroups.length >= 3, "Not enough groups for test setup");
+    ctx.targetGroup = initialGroups[2].group;
```

Comment:
No idea how value `2` was invented.
Is there some assumption about number of groups in test configuration or setup?

Reply:
In most test cases we have 3 validator groups. I randomly chose one of them :-)

Reply:
Some small idea might be than either to add a comment with this info or maybe use constant with description of intention 😄